Modules/_asynciomodule.c

```diff
@@ -2754,7 +2755,17 @@ task_step_impl(TaskObj *task, PyObject *exc)
             _PyErr_StackItem *exc_state = &fut->fut_cancelled_exc_state;
             exc_state->exc_value = ev;
 
-            return future_cancel(fut, NULL);
+            args = ((PyBaseExceptionObject*)ev)->args;
+            if (PyTuple_Size(args) > 0) {
+                PyObject *msg = PyTuple_GetItem(args, 0);
+                if (msg == NULL) {
+                    return NULL;
+                }
+                return future_cancel(fut, msg);
+            }
+            else {
+                return future_cancel(fut, NULL);
+            }
```

Comment:
Use macros, and you will not need to check for error.
```suggestion
            PyObject *msg = PyTuple_GET_SIZE(args) ? PyTuple_GET_ITEM(args, 0) : NULL;
            return future_cancel(fut, msg);
```